File: Project/WebService/Services/DTOs/RatingDTO.cs
// --------------------------------------------------------------------------------------------------------------------
// <copyright file="RatingDTO.cs" company="">
//   
// </copyright>
// <summary>
//   Defines the RatingDTO type.
// </summary>
// --------------------------------------------------------------------------------------------------------------------

namespace Services.DTOs
{
    using System.Runtime.Serialization;

    /// <summary>
    /// The rating dto.
    /// </summary>
    [DataContract]
    public class RatingDTO
    {
        /// <summary>
        /// Gets or sets the rating id.
        /// </summary>
        [DataMember]
        public int RatingId { get; set; }

        /// <summary>
        /// Gets or sets the user id.
        /// </summary>
        [DataMember]
        public int TargetUserId { get; set; }

        /// <summary>
        /// Gets or sets the from user id.
        /// </summary>
        [DataMember]
        public int FromUserId { get; set; }

        /// <summary>
        /// Gets or sets the score.
        /// </summary>
        [DataMember]
        public int Score { get; set; }

        /// <summary>
        /// Gets or sets the feedback.
        /// </summary>
        [DataMember]
        public string Feedback { get; set; }
    }
}

File: Project/Client/MyApplicationProject/MyApplication/src/main/java/com/example/myapplication/dtos/RatingDTO.java
package com.example.myapplication.dtos;

/**
 * Used when a user submits a new rating for driver of one of the journeys they participated in.
 */
public class RatingDTO {
    private int TargetUserId;
    private int FromUserId;
    private int Score;
    private String Feedback;

    public RatingDTO(int targetUserId, int score, int fromUserId, String feedback) {
        TargetUserId = targetUserId;
        Score = score;
        FromUserId = fromUserId;
        Feedback = feedback;
    }
}
